Author: Balazs Gibizer <gibi@redhat.com>
Date: Wed, 15 Jun 2022 13:13:20 +0200
Description: [PATCH] Make test_wait_for_instance_event_* test time independent
 I91c42af8226dc8607d86c52413924351ba0586ac introduced three test cases
 that unfortunately dependent on execution time. The code under test uses
 a StopWatch  to measure execution time and include that in a log that is
 then assert by the test cases. This cases the test assertion of "0.00
 seconds" to fail on a slow CI node.
 .
 This patch removes the dependency towards StopWatch.elapsed to stabilize
 the tests.
Bug-Ubuntu: https://launchpad.net/bugs/1978817
Change-Id: Id1ba3a886cf2d31956fc8f51dc7de08a79b22922
Bug-Debian: https://bugs.debian.org/1013410
Origin: upstream, https://review.opendev.org/c/openstack/nova/+/845922
Last-Update: 2022-06-30

diff --git a/nova/tests/unit/compute/test_virtapi.py b/nova/tests/unit/compute/test_virtapi.py
index d6dc8f1..c56b2cf 100644
--- a/nova/tests/unit/compute/test_virtapi.py
+++ b/nova/tests/unit/compute/test_virtapi.py
@@ -187,6 +187,9 @@
 
         do_test()
 
+    @mock.patch(
+        'oslo_utils.timeutils.StopWatch.elapsed',
+        new=mock.Mock(return_value=1.23))
     def test_wait_for_instance_event_timeout(self):
         instance = mock.Mock()
         instance.vm_state = mock.sentinel.vm_state
@@ -212,11 +215,14 @@
                 'vm_state': mock.sentinel.vm_state,
                 'task_state': mock.sentinel.task_state,
                 'event_states':
-                    'foo-bar: timed out after 0.00 seconds',
+                    'foo-bar: timed out after 1.23 seconds',
             },
             instance=instance
         )
 
+    @mock.patch(
+        'oslo_utils.timeutils.StopWatch.elapsed',
+        new=mock.Mock(return_value=1.23))
     def test_wait_for_instance_event_one_received_one_timed_out(self):
         instance = mock.Mock()
         instance.vm_state = mock.sentinel.vm_state
@@ -252,12 +258,15 @@
                 'vm_state': mock.sentinel.vm_state,
                 'task_state': mock.sentinel.task_state,
                 'event_states':
-                    'foo-bar: received after waiting 0.00 seconds, '
-                    'missing-event: timed out after 0.00 seconds',
+                    'foo-bar: received after waiting 1.23 seconds, '
+                    'missing-event: timed out after 1.23 seconds',
             },
             instance=instance
         )
 
+    @mock.patch(
+        'oslo_utils.timeutils.StopWatch.elapsed',
+        new=mock.Mock(return_value=1.23))
     def test_wait_for_instance_event_multiple_events(self):
         instance = mock.Mock()
         instance.vm_state = mock.sentinel.vm_state
@@ -282,7 +291,6 @@
             m.tag = tag
             m.event_name = '%s-%s' % (name, tag)
             m.wait.side_effect = fake_event_waiter
-            print(name, tag)
             if name == 'received-but-not-waited':
                 m.ready.return_value = True
             if name == 'missing-but-not-waited':
@@ -323,9 +331,9 @@
                 'vm_state': mock.sentinel.vm_state,
                 'task_state': mock.sentinel.task_state,
                 'event_states':
-                    'received-event: received after waiting 0.00 seconds, '
+                    'received-event: received after waiting 1.23 seconds, '
                     'early-event: received early, '
-                    'missing-event: timed out after 0.00 seconds, '
+                    'missing-event: timed out after 1.23 seconds, '
                     'received-but-not-waited-event: received but not '
                     'processed, '
                     'missing-but-not-waited-event: expected but not received'
